### ISSUE TITLE
Proposing BUNDLE DIRECTORY

### ISSUE THREAD

**Issue by felixbur** [felixbur/nkululeko#373]
from @bagustris 

Proposing BUNDLE DIRECTORY

Nowadays, people share their models freely on Google Drive, Hugging Face, or other sharing platforms. We cannot do the same for Nkululeko. Hence, I propose a bundle directory so everyone can use the model generated by Nkululeko without training it. Assumed that the user only needs to perform the two steps below:

    Install nkululeko
    Download the BUNDLE DIRECTORY (model file, usually also with model scaler, afaik, for a scikit-learn-based model, config file, and other required files)

And then run it with:

```
# if no config file is given for demo module, then next value will be treated as bundle directory
# or perhaps if no subcommand given after infer module, it will treat next value as bundle directory
python -m nkululeko.infer demo [BUNDLE DIRECTORY] --file new_file.wav
# or
python -m nkululeko.infer [BUNDLE DIRECTORY] --file new_file.wav
```

Where BUNDLE DIRECTORY contains required files (model, modified config file, feature extractor, scaler, label encoder, metadata, etc). Next step, we need to add a new key in the config file to generate the BUNDLE DIRECTORY:

```
[EXPORT] 
bundle = True # default to false (or make default true to solve problem below?)
path = "/path/to/export  # default to <root>/<name>/export
```

Some concerns:

* Sharing the config file with the public may leak some important information, particularly if using internal data. So, generate bundle files that are only required for inference.
* We usually want to share the model after knowing that the results are good, so perhaps we need an additional command to only generate bundle files after training ends (or use bundle true to avoid this).

**Comment by bagustris** [comment]
Hi @felixbur, thank you for copying my previous issue to here.

I'll ask copilot to implement with specification below.

```
  my_model_bundle/
    manifest.json
    inference.ini
    model.pkl
    scaler.pkl
    label_encoder.pkl
    feature_schema.json
    requirements.txt
    metrics.json
    README.md
```
  The key file is manifest.json, for example:

```json
  {
    "bundle_version": "1.0",
    "nkululeko_version": "x.y.z",
    "python_version": "3.11",
    "task": "classification",
    "target": "emotion",
    "labels": ["anger", "joy", "neutral", "sadness"],
    "features": {
      "type": ["bert", "os"],
      "os.set": "eGeMAPSv02",
      "bert.model": "bert-base-uncased",
      "scale": "standard"
    },
    "artifacts": {
      "model": "model.pkl",
      "scaler": "scaler.pkl",
      "label_encoder": "label_encoder.pkl",
      "inference_config": "inference.ini",
      "feature_schema": "feature_schema.json"
    }
  }
```

  The important design choice: do not export the original training config
  directly. Generate a sanitized inference.ini that keeps only what inference
  needs:

```ini
  [EXP]
  language = en

  [DATA]
  target = emotion
  labels = ['anger', 'joy', 'neutral', 'sadness']

  [FEATS]
  type = ['bert', 'os']
  bert.model = bert-base-uncased
  os.set = eGeMAPSv02
  scale = standard

  [MODEL]
  type = svm
```

  This addresses the privacy concern in issue above because training paths,
  internal dataset names, split strategies, and result directories do not need
  to be shared.

  The CLI proposed in the issue makes sense:

      python -m nkululeko.infer my_model_bundle --file new_file.wav

  I would implement it in two phases:

  1. MVP (minimum viable product) export
      - Add [EXPORT] bundle = True.
      - After training, export best model, label encoder, scaler, sanitized
        config, metadata.
      - Add `python -m nkululeko.infer BUNDLE --file audio.wav`.
  2. Robust bundle
      - Validate Nkululeko/Python/dependency versions.
      - Validate feature dimension/order with feature_schema.json.
      - Support Hugging Face style upload/download later.
      - Optionally support ONNX for compatible sklearn models.

Short question @felixbur: did you want this feature to be integrated with `predict` module or separate `infer` module?

**Comment by felixbur** [comment]
I'd prefer an extra module,, predict already is quite large
